src/components/Table/hoc/withTableSettings/withTableSettings.tsx

```diff
@@ -159,51 +154,33 @@ export function withTableSettings<I extends TableDataItem, E extends {} = {}>(
             settingsPopupWidth,
             ...restTableProps
         }: TableProps<I> & WithTableSettingsProps & E) {
-            const actualItems = React.useMemo(
-                () => getActualItems(columns, settings || []),
-                [columns, settings],
-            );
-
-            const onUpdateColumns = React.useCallback(
-                (newItems: TableColumnSetupItem[]) => {
-                    updateSettings(prepareUpdateSettings(newItems));
-                },
-                [updateSettings],
-            );
-
-            const columnSetupItems = React.useMemo(
-                () => prepareColumnSetupItems(actualItems),
-                [actualItems],
-            );
-
-            const enhancedColumns = React.useMemo(
-                () =>
-                    enhanceSystemColumn(filterColumns(columns, actualItems), (systemColumn) => {
-                        // eslint-disable-next-line react/display-name
-                        systemColumn.name = () => (
-                            <div className={b('settings')}>
-                                <TableColumnSetup
-                                    popupWidth={settingsPopupWidth || width}
-                                    popupPlacement={['bottom-end', 'bottom', 'top-end', 'top']}
-                                    sortable={sortable}
-                                    onUpdate={onUpdateColumns}
-                                    items={columnSetupItems}
-                                    renderSwitcher={({onClick}) => (
-                                        <Button
-                                            view="flat"
-                                            className={b('settings-button')}
-                                            extraProps={{'aria-label': i18n('label_settings')}}
-                                            onClick={onClick}
-                                        >
-                                            <Icon data={Gear} />
-                                        </Button>
-                                    )}
-                                />
-                            </div>
-                        );
-                    }),
-                [actualItems, columnSetupItems, columns, onUpdateColumns, settingsPopupWidth],
-            );
+            const enhancedColumns = React.useMemo(() => {
+                const actualItems = getActualItems(columns, settings || []);
+
+                return enhanceSystemColumn(filterColumns(columns, actualItems), (systemColumn) => {
+                    systemColumn.name = () => (
+                        <div className={b('settings')}>
+                            <TableColumnSetup
+                                popupWidth={settingsPopupWidth || width}
```

Comment:
1) Whats the difference between `settingsPopupWidth` and `width`? Could we stick to only one setting for this value?

2) For now default popup width is too wide.

Reply:
1 - honestly, I do not know. [It was](https://github.com/gravity-ui/uikit/blob/main/src/components/Table/hoc/withTableSettings/withTableSettings.tsx#L186) before I came. 

2 - seems `TreeSelect` can not how to `width === fit-content`.

Reply:
1. Oh, I see, it's deprecated property, but we still didn't remove it.
2. I see that `TreeSelect` has `min-width: 300px`. What if we pass `popupWidth=auto` to it?

Reply:
2 - nothing will happen, the same width